tests/sentry/rules/processing/test_delayed_processing.py

```diff
@@ -142,9 +143,66 @@ def test_build_group_to_groupevent(self):
         pass
 
 
-class BulkFetchEventsTest(TestCase):
-    def test_bulk_fetch_events(self):
-        pass
+class BulkFetchEventsTest(CreateEventTestCase):
+    def setUp(self):
+        super().setUp()
+        self.project = self.create_project()
+        self.environment = self.create_environment(project=self.project)
+
+        self.event = self.create_event(
+            self.project.id, FROZEN_TIME, "group-1", self.environment.name
+        )
+        self.event_two = self.create_event(
+            self.project.id, FROZEN_TIME, "group-1", self.environment.name
+        )
+
+        self.expected = {
+            self.event.event_id: self.event,
+            self.event_two.event_id: self.event_two,
+        }
+
+    def test_basic(self):
+        event_ids: list[str] = [self.event.event_id, self.event_two.event_id]
+        result = bulk_fetch_events(event_ids, self.project.id)
+
+        assert len(result) == len(self.expected)
+        for expected, fetched in zip(self.expected, result):
+            assert expected == fetched
+
+    def test_with_invalid_event_id(self):
+        event_ids: list[str] = ["0", "1"]
```

Comment:
do we know with certainty these will never be the correct event ids?

Reply:
`0` yes because of the db schema, but `1` no.. i'll update to be `-1` for that case as well as the others. 👍